Source/Core/DolphinQt/Config/HardcoreWarningWidget.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#ifdef USE_RETRO_ACHIEVEMENTS
+#include "DolphinQt/Config/HardcoreWarningWidget.h"
+
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QPixmap>
+#include <QPushButton>
+#include <QStyle>
+
+#include "Core/Config/AchievementSettings.h"
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+
+#include "DolphinQt/Settings.h"
+
+HardcoreWarningWidget::HardcoreWarningWidget(QWidget* parent) : QWidget(parent)
+{
+  CreateWidgets();
+  ConnectWidgets();
+
+  connect(&Settings::Instance(), &Settings::EmulationStateChanged, this, [this] { Update(); });
+
+  Update();
+}
+
+void HardcoreWarningWidget::CreateWidgets()
+{
+  auto* icon = new QLabel;
```

Comment:
Maybe move this down to above the setPixmap call, so it's declaration and first use are close by.